### ISSUE TITLE
Global reduced motion config

### ISSUE THREAD

**Issue by JacksonKearl** [microsoft/vscode#128595]
On MacOS (and probably windows too), there's a global "reduce motion" configuration. It would be nice if we could have something similar, possibly using that global value by default. 

![image](https://user-images.githubusercontent.com/8586769/125542314-178a4b9d-6ef1-4727-9bb9-f268b5158be1.png)

**Comment by JacksonKearl** [comment]
Started to implement this with f76d07f, using `workbench.welcomePage.preferReducedMotion`

**Comment by isidorn** [comment]
I think this makes sense. However we should have a global workbench setting, and not a setting per area. So the user could just flip one setting, and it is up to various workbench areas to respect this setting. Due to that I suggest to name this
`workbench.reduceMotion`. And in the setting description to explain that this is only supported for now by the welcome page...
`preferReducedMotion` is not very concise and I am not sure what `prefer` brings?

**Comment by isidorn** [comment]
If we can already get this from the OS as @bpasero says maybe we should just respect it out of the box and we do not need a setting at all. Only in the future if users want to disable this automatic respecting of OS setting we can introduce a setting.
I guess we can not get this automatically from the browser? So when in web a setting might make sense.

So I suggest

`workbench.reduceMotion: "auto" | "on" | "off"`

`auto` is the default and that means pick ip up from Electron API. For web users would have to explicitly set to `"on"`

**Comment by bpasero** [comment]
I think you will always find users that want a setting if you wait long enough. This is probably similar to the settings we have around detecting high contrast theme or dark mode in the OS where we have a setting to override this.

**Comment by JacksonKearl** [comment]
Added `reduce-motion` / `enable-motion` classes to the base `.monaco-workbench` and some new bits on `IAccessibilityService`:
```ts
readonly onDidChangeReducedMotion: Event<void>;
isMotionReduced(): boolean;
```

**Comment by isidorn** [comment]
@JacksonKearl this is a great start. 
I think we can disable all usages of animations in css when motion is reduced. A good way to find animation is to search for "keyframe" or "animation-name" in our repository. On top of that we could change the cursor to `"editor.cursorBlinking": "solid"`

fyi @bpasero @joaomoreno for animations we use

I just tried Apple reduce motion and they do not change mouse hover interaction. So I suggest we keep that.
I will consult accessibility team if there are other things we should disable.

**Comment by JacksonKearl** [comment]
Interesting, from Apple's [Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/accessibility/overview/appearance-effects/):

> Be cautious when displaying moving or blinking elements. Although subtle movement and blinking can draw the user’s attention, these effects can also be distracting and they aren’t useful for people with visual disabilities. Worse, some blinking elements can cause epileptic episodes. In all cases, avoid using movement and blinking as the only way to convey information.

But then they don't disable blinking cursors. Also, with system wide Reduce Motion enabled, I still see blinking cursors everywhere. I think it makes good sense to edit the motion reducing text to include "animation" for queries though, that's what many related settings use already.

**Comment by isidorn** [comment]
@JacksonKearl thanks for looking into Apple's guidelines.
I have reached out to @dantegagne and he informed me that Windows has a setting for disabling cursor blink - it is in the old Keyboard Properties section in Control Panel, so it’s not particularly discoverable, but it’s there.  VS respects this setting. Also VS does not have a lot of animations so there is not much to disable for them.

@JacksonKearl I guess that on Windows we can not automatically detect if the user disabled the cursor blink?
I leave it up to you if turning this on should disable cursor blinking. I am leaning towards yes, but we can fine tune after we get feedback.

![image](https://user-images.githubusercontent.com/1926584/157457388-4f07cd48-efb4-4a07-9199-b8cdafc88c08.png)
